reconcile/openshift_resources_base.py

```diff
@@ -1357,9 +1367,6 @@ def desired_state_shard_config() -> DesiredStateShardConfig:
     return DesiredStateShardConfig(
         shard_arg_name="cluster_name",
         shard_arg_is_collection=True,
-        shard_path_selectors={
-            "namespaces[*].cluster.name",
-            "resources[*].cluster",
-        },
+        shard_path_selectors={"state.*.shard"},
```

Comment:
Can this be even `{"state.*"}` given the state key is the shard name/id? Then we would not even need teh `"shard"` key in the value. The value could  even be directly the sha?
```
return {
        "state": {
            cluster: DeepHash(state).get(state)
            for cluster, state in state_for_clusters.items()
        }
    }
```
Or would the diff mechanism not work with that?

Reply:
I tried that but with jsonpath there is no way to match and extract on a dict key, only values. 

But in the end it does not matter. I see a simplification refactor upcoming where we can get rid of jsonpath altogether by coming up with a standard to structure the desired state.